File: cpp_solutions/1189. Maximum Number of Balloons.cpp
//ss
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        
        vector<int> maps(26,0);
        
        for (char ch: text){
            maps[int(ch)-int('a')]+=1;
        }
        
        int ans = 10000;
        // b , a , l/2, o/2, n -> mins
        string temp = "balon";
        
        for (char ch: temp){
            if (ch == 'b' || ch == 'a' || ch == 'n'){
                ans = min(ans, maps[int(ch)-int('a')]);
            }
            else{
                ans = min(ans, (int)(maps[int(ch)-int('a')]/2));
            }
        }
        return ans;
        
    }
};

File: python3_solution_set0/1189. Maximum Number of Balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        
        ans = [0,0,0,0,0]
        
        for x in range(len(text)):
            if text[x]=='b':
                ans[0]+=1
                
            elif text[x]=='a':
                ans[1]+=1
                
            elif text[x]=='l':
                ans[2]+=1
                
            elif text[x]=='o':
                ans[3]+=1
                
            elif text[x]=='n':
                ans[4]+=1
                
                
        if ans[2]%2==0:
            ans[2] = ans[2]/2
            
        else:
            ans[2] = int((ans[2]-1)/2)
            
            
        if ans[3]%2==0:
            ans[3] = ans[3]/2
            
        else:
            ans[3] = int((ans[3]-1)/2)
            
            
        return int(min(ans))
        
